src/common/feedGenerator.ts

```diff
@@ -113,10 +113,19 @@ export const feedToFilters = async (
       })
       .execute(),
     feedId
-      ? con.getRepository(SourceMember).find({
-          where: { userId },
-          select: ['sourceId'],
-        })
+      ? con
+          .getRepository(SourceMember)
+          .createQueryBuilder('sm')
+          .select('sm."sourceId"')
+          .where('sm."userId" = :userId', { userId })
+          .andWhere(
+            new Brackets((qb) => {
+              qb.where(
+                `(sm."flags"->'showPostsOnFeed')::boolean = TRUE`,
+              ).orWhere(`sm."flags"->'showPostsOnFeed' IS NULL`);
```

Comment:
Had to do this kind of query because if `flags.showPostsOnFeed` on feed is not set the query `WHERE (flags->'showPostsOnFeed')::boolean != false` would not return correctly, it just does not work with nullish values for json paths. That is why I have `OR` condition to also check for `NULL`.

> Why did we not just set `flags.showPostsOnFeed` to `true` by default?

During initial post flags PR https://github.com/dailydotdev/daily-api/pull/1333#discussion_r1259503040 we figured that if any of our flags have value (like true by default) we would need this kind of trigger to actually preserve those default during insert.

Reply:
A better approach would be to reverse the flag. So as we discussed in the post table flags, it's better to have flags that are false/null by default. So maybe we should call it `hideFromFeed`, and only if it's true we will not show it on the feed. May reduce the edge cases here, right?
Some lessons learned from previous migrations

Reply:
@idoshamun I thought about that but it would not make a difference in regards to the query, we would still need `NOT NULL` check since our `flags->hideFromFeed` would still not have value by default and query `WHERE (flags->'hideFromFeed')::boolean != true` would still not return matches for all those source members.

Reply:
@capJavert have you tried using the `COALESCE` function? `COALESCE(flags->'hideFromFeed', false)`

Reply:
@idoshamun nice one, looks like it works, gonna refactor to `hideFromFeed` first thing in the morning so we can close this.